File: java-server/src/main/java/com/city/javaserver/DummyObject.java
package com.city.javaserver;


import lombok.*;
import org.bson.BSONObject;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class DummyObject implements Serializable {
    private String id;
    private int index;
    private String guid;
    private Boolean isActive;
    private double balance;
    private String picture;
    private int age;
    private String eyeColor;
    private String name;
    private String gender;
    private String company;
    private String email;
    private String phone;
    private String address;
    private String about;
    private String registered;
    private double latitude;
    private double longitude;
    private String[] tags;
    private DummyFriend[] friends;
    private String greeting;
    private String favoriteFruit;
}

File: rust_server/src/structs/dummy_object.rs
use serde::{Deserialize,Serialize};
use crate::structs::dummy_friend::DummyFriend;
#[derive(Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DummyObject {
    pub id: String,
    pub index: u8,
    pub guid: String,
    pub is_active: bool,
    pub balance: f32,
    pub picture: String,
    pub age: u8,
    pub eye_color: String,
    pub name: String,
    pub gender: String,
    pub company: String,
    pub email: String,
    pub phone: String,
    pub address: String,
    pub about: String,
    pub registered: String,
    pub latitude: f64,
    pub longitude: f64,
    pub tags: Vec<String>,
    pub friends: Vec<DummyFriend>,
    pub greeting: String,
    pub favorite_fruit: String,
}
